### PR TITLE
Bash tab completions for subcommands swarm, service and node

### PR DESCRIPTION
**Pull request by ghost**
Adds bash tab completion for `swarm`, `service` and `node` subcommands.

Includes some basic intelligence about the use of these commands.  For example, `docker node accept` will only complete to pending node names/ids.

@albers Let me know if you're interested in this contribution.

### CODE REVIEW

**Comment by albers** [comment]
Thanks, @mgoelzer, quite a big change! 
After a quick glance, this looks very well.
I will review it once the corresponding functionality is merged.
(There might be changes in those PRs that might require rework, and it's generally a good idea to avoid dependencies between PRs as it makes reviewing harder.)
Can you please add links to the PRs that implement the functionality?

**Comment by mgoelzer** [comment]
Thank you, @albers.  

This is the (currently unmerged) PR that changes the `docker` CLI to add new subcommands commands:  #23363

Agree with you that this will need updates as #23363 changes.  I'm watching it.

**Comment by vieux** [comment]
@albers it was just merged :)

**Comment by albers** [comment]
@vieux Thanks for the ping.

Please give me several days to review. I will have to understand all the new features (which sometimes turns out to be quite cumbersome as there can be small nits/inconsistencies in the documentation or the features themselves).

**Comment by albers** [comment]
@mgoelzer Is this up to date with the final state of #23363?

**Comment by mgoelzer** [comment]
@albers  Yes, it is up to date with #23363.

**Comment by vieux** [comment]
LGTM

File: contrib/completion/bash/docker [hunk 1]
```diff
@@ -1465,6 +1514,246 @@ _docker_network() {
...
+		*)
+			COMPREPLY=( $( compgen -W "$subcommands" -- "$cur" ) )
+			;;
+	esac
+}
+
+_docker_service_create() {
+	case "$cur" in
+		-*)
+			COMPREPLY=( $( compgen -W "--help --constraint --endpoint-ingress --endpoint-mode --env --label --limit-cpu --limit-memory --mode --name --network --publish --reserve-cpu --reserve-memory --restart-condition --restart-delay --restart-max-attempts --restart-window --scale --stop-grace-period --update-delay --update-parallelism --user --volume --workdir" -- "$cur" ) )
```
Comment (vieux) [inline contrib/completion/bash/docker]: `--scale` should be `--replicas`

File: contrib/completion/bash/docker [hunk 2]
```diff
@@ -1465,6 +1514,246 @@ _docker_network() {
...
+			;;
+	esac
+}
+
+_docker_service_ls() {
+	case "$cur" in
+		-*)
```
Comment (vieux) [inline contrib/completion/bash/docker]: ls doesn't have all those arguments. I thinks `update` and `ls` were inverted
`--scale` should be `--replicas`

**Comment by mgoelzer** [comment]
Thank you for the feedback @vieux.  The problems you noticed are fixed.

**Comment by mlaventure** [comment]
@mgoelzer it needs a rebase

**Comment by mgoelzer** [comment]
@mlaventure Thanks -- it's rebased now.

**Comment by mlaventure** [comment]
LGTM (IANAM)

**Comment by vieux** [comment]
LGTM

**Comment by thaJeztah** [comment]
LGTM

@albers if you have corrections on this, please let us know then we can include them in a follow up PR before the next RC :heart:

**Comment by albers** [comment]
I hate to say this, but It was no good idea to merge this so quickly. 
There are some issues: all options are treated as booleans, there are options missing or non existent. The one-letter options are missing, command aliases are not covered.
I will create follow-up PRs.

**Pull request merged**